oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java

```diff
@@ -684,6 +691,13 @@ public int getMemory() {
 
         diffCache = builder.getDiffCache(this.clusterId);
 
+        final Feature prevNoPropFeature = builder.getPrevNoPropCacheFeature();
+        if (prevNoPropFeature == null || !prevNoPropFeature.isEnabled()) {
+            prevNoPropCache = null;
+        } else {
+            prevNoPropCache = builder.buildPrevNoPropCache(this);
+        }
```

Comment:
I would move this to [Utils](https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/util/Utils.java)

Reply:
good point, will do

Reply:
moved to the builder actually, as a few things need to be checked, done in https://github.com/apache/jackrabbit-oak/pull/1779/commits/faa44dffb661a8776dcef7d335e239096e17fce4

(will add some more unit tests)